app/Main.hs

```diff
@@ -93,25 +95,24 @@ main = (recreateConfig . getOpt Permute opts <$> getArgs) >>= \case
     forever $ tryPutMVar upd () >> threadDelay (view cInterval cfg)
 
   updater upd viewRef cfg =
-    forever
-      $ withFDHandler defaultBatchingOptions stdout 1.0 80
-      $ \logCallback -> runLoggingT
-          (runReaderT (refresh viewRef upd) cfg)
-          ( logCallback
-          . renderWithTimestamp
-            (formatTime defaultTimeLocale "T%H:%M:%S")
-            id
-          )
+    forever .
+    runReader cfg .
+    withStdOutLogger $ \logger ->
+                         runLog "main" logger defaultLogLevel
+                         (refresh viewRef upd)
 
 webserver
-  :: Config
+  :: ( IOE :> es
+     , Concurrent :> es
+     )
+  => Config
   -> IORef View -- ^ View model
   -> MVar () -- ^ Update signal
-  -> IO ()
+  -> Eff es ()
 webserver Config{_cPort=webserverPort} viewRef upd =
-  run webserverPort $ foldr
+  liftIO . run webserverPort $ foldr
     (logStdout . static $(embedDir "static") <$>)
     (notFound (send $ html "not found..."))
     [ get "/"  (liftIO (readIORef viewRef) >>= send . html . encodeUtf8 . render)
-    , get "/r" (liftIO (tryPutMVar upd ()) >> send (redirect302 "/"))
+    , get "/r" (liftIO (runEff . runConcurrent $ tryPutMVar upd ()) >> send (redirect302 "/"))
     ]
```

Comment:
`webserver` looks a bit ugly, if someone wants to figure it out, be my guest

Reply:
I gave it a go and couldn't make it less ugly. What makes it ugly btw? :joy_cat:

Reply:
It's mostly that warp's `run` is of type `Port -> Application -> IO ()`, and that `(liftIO (runEff . runConcurrent $ tryPutMVar upd ()) >> send (redirect302 "/"))` has type `ResponderM`. This means that we have to run `runEff . runConcurrent $ tryPutMVar upd ()` to access the MVar from Effectful.

One potential fix is to extend warp and twain, but that seems like a lot of work

Reply:
Yeah, that's not super beautiful. It's like a cake with `liftIO` sandwiched between almost every layer. I might have another variant of this code for you, let me put it in a commit. 

Extending warp and twain would be exciting but perhaps a bit too ambitious, especially in this PR. :smile_cat: 

Another ambitious route is to use Servant instead and make things even more typed! Here's the box of Pandora: https://docs.servant.dev/en/stable/cookbook/hoist-server-with-context/HoistServerWithContext.html

Reply:
After some experimenting back and forth I came to the conclusion that I do not have another variant. :smile: